envi/codeflow.py

```diff
@@ -230,23 +230,26 @@ def addCodeFlow(self, va, arch=envi.ARCH_DEFAULT):
                                 # descend into functions, but make sure we don't descend into
                                 # recursive functions
                                 if bva in self._cf_blocks:
+                                    logger.debug("self._cf_recurse == True, but 0x%x is in self._cf_blocks (ie. it called *this* func)", bva)
+                                    logger.debug("\t" + ", ".join([hex(x) for x in self._cf_blocks]))
                                     # the function that we want to make prodcedural
                                     # called us so we can't call to make it procedural
                                     # until its done
                                     cf_eps[bva] = bflags
                                 else:
+                                    logger.debug("self._cf_recurse == True, so we're descending into 0x%x", bva)
```

Comment:
when identifying why locations are wrong (and we do have quite a few suspects to fix), this can prove beneficial as delineators of what's going on, providing context for the creation of the location that's wrong.

i *hope* we see this a lot, we would have never seen it before now (can you tell i'm still giddy about that?)... but methinks we'll see it a lot less than you think.  and i'm pretty sure you're going to want to steer clear of debug and below messages.  i think we should go through and make sure the things you're interested in show up in INFO.  we do that and i think we'll have a pretty good starting hierarchy for log messages.

Reply:
i will mod the message, tho.  that's super specific code....